plugins/modules/ec2_spot_instance_info.py

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/python
+# This file is part of Ansible
+# GNU General Public License v3.0+ (see COPYING or https://wwww.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = '''
+---
+module: ec2_spot_instance_info
+version_added: 2.0.0
+short_description: Gather information about ec2 spot instance requests
+description:
+  - Describes the specified Spot Instance requests.
+author:
+  - Mandar Vijay Kulkarni (@mandar242)
+options:
+  filters:
+    description:
+      - A list of filters to apply.
+      - Each list item is a dict consisting of a filter key and a filter value.
+      - See U(https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeSpotInstanceRequests.html) for possible filters.
+    type: list
+    elements: dict
+  spot_instance_request_ids:
+    description:
+      - One or more Spot Instance request IDs.
+    type: list
+    elements: str
+
+extends_documentation_fragment:
+- amazon.aws.aws
+- amazon.aws.ec2
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+- name: describe the Spot Instance requests based on request IDs
+  amazon.aws.ec2_spot_instance_info:
+    spot_instance_request_ids:
+      - sir-12345678
+
+- name: describe the Spot Instance requests and filter results based on instance type
+  amazon.aws.ec2_spot_instance_info:
+    spot_instance_request_ids:
+      - sir-12345678
+      - sir-13579246
+      - sir-87654321
+    filters:
+      - name: 'launch.instance-type'
+        values:
+          - t3.medium
+
+- name: describe the Spot requests filtered using multiple filters
+  amazon.aws.ec2_spot_instance_info:
+    filters:
+      - name: 'state'
```

Comment:
We simplify filters for users so that they don't have to put `name` and `values` as keys.  `module_utils.ec2` has a helper `ansible_dict_to_boto3_filter_list` that will do this for you.  That way users can just say:
```
filters:
  - state: active
```